backend/pyi_rth_rocm_sdk.py

```
@@ -0,0 +1,81 @@
+    import platform
+    from pathlib import Path
+
+    def find_libraries(*shortnames):
+        paths = []
+        missing_extras = set()
+        is_windows = platform.system() == "Windows"
+        for shortname in shortnames:
+            try:
+                lib_entry = _dist_info.ALL_LIBRARIES[shortname]
+            except KeyError:
+                raise ModuleNotFoundError(f"Unknown rocm library '{shortname}'")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)